tests/utils/basic-skipif.inc

```diff
@@ -1,9 +1,24 @@
 <?php
 
-if ( ! extension_loaded("mongodb")) {
-    exit("skip mongodb not loaded");
+require_once __DIR__ . '/basic.inc';
+require_once __DIR__ . '/skipif.php';
+
+if ( ! extension_loaded('mongodb')) {
+    exit('skip mongodb extension is not loaded');
 }
 
-require __DIR__ . "/" . "basic.inc";
+set_error_handler(function($errno, $errstr) {
+    exit(sprintf('skip %s: %s', errno_as_string($errno), $errstr));
+});
+
+set_exception_handler(function($e) {
+    exit(sprintf('skip %s(%d): %s', get_class($e), $e->getCode(), $e->getMessage()));
+});
 
+register_shutdown_function(function() {
```

Comment:
See: http://php.net/manual/en/function.set-error-handler.php#112291

At the moment, I'm using all of these handlers to catch errors during `SKIPIF` refactoring. We can decide if they're worth keeping later on. It may be preferable to ignore `SKIPIF` errors/exceptions entirely and allow a test to fail outright.